### PR TITLE
Fixes for NoOpTransportProvider

### PR DESCRIPTION
**Pull request by jzakaryan**
We'll need these fixes for `NoOpTransportProvider` if we want to use it in production.

Please, take a minute to review the contribution guidelines at:  
https://github.com/linkedin/Brooklin/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java [hunk 1]
```diff
@@ -60,6 +60,7 @@ public class EventProducer implements DatastreamEventProducer {
   static final String EVENTS_SEND_LATENCY_MS_STRING = "eventsSendLatencyMs";
   static final String THROUGHPUT_VIOLATING_EVENTS_LATENCY_MS_STRING = "throughputViolatingEventsLatencyMs";
   static final String THROUGHPUT_VIOLATING_EVENTS_SEND_LATENCY_MS_STRING = "throughputViolatingEventsSendLatencyMs";
+  static final String NOOP_TRANSPORT_PROVIDER_NAME = "NoOp";
```
Comment (vmaheshw) [inline datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java:63]: Can this string constant be reused?

File: datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java [hunk 2]
```diff
@@ -271,8 +272,11 @@ private DatastreamRecordMetadata helperSendOrBroadcast(DatastreamProducerRecord
 
     try {
       // Send the event to the transport
-      String destination =
-          record.getDestination().orElse(_datastreamTask.getDatastreamDestination().getConnectionString());
+      String destination = "dummy";
```
Comment (vmaheshw) [inline datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java:275]: Is `dummy` destination okay for NOOP Transport provider?

File: datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java (hunk 1 above)
Comment (jzakaryan) [reply datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java]: Done. Moved the constant to `NoOpTransportProvider` class.

**Review by vmaheshw — APPROVED** [review]

File: datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java (hunk 2 above)
Comment (jzakaryan) [reply datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java]: It is OK for NoOp TransportProvider (it doesn't lookup the destination string). Changed it to an empty string constant to make it more pleaseant.

**Review by atoomula — APPROVED** [review]

**Pull request merged**